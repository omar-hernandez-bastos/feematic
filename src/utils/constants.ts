export enum ActionsPad {
  CLEAR = 'ðŸš«',
  DECIMAL = '.'
}
export const pads = [
  { text: 'ðŸ‡¨ðŸ‡´', className: '', color: 'cyan-200' },
  { text: 'ðŸ‡ºðŸ‡¸', className: '', color: 'gray-100' },
  { text: 'ðŸ‡»ðŸ‡ª', className: 'row-span-2' },
  { text: '7', className: '' },
  { text: '8', className: '' },
  { text: '9', className: '' },
  { text: '4', className: '' },
  { text: '5', className: '' },
  { text: '6', className: '' },
  { text: '1', className: '' },
  { text: '2', className: '' },
  { text: '3', className: '' },
  { text: ActionsPad.CLEAR, className: 'row-span-2' },
  { text: '0', className: 'col-span-2' },
  { text: ActionsPad.DECIMAL, className: '' }
]

export type CURRENCIES = 'ðŸ‡¨ðŸ‡´' | 'ðŸ‡ºðŸ‡¸' | 'ðŸ‡»ðŸ‡ª'
const currencies = {
  COP: 'ðŸ‡¨ðŸ‡´',
  USD: 'ðŸ‡ºðŸ‡¸',
  VES: 'ðŸ‡»ðŸ‡ª'
}
export const CURRENCIES_ARRAY = [currencies.USD, currencies.COP, currencies.VES]
export { currencies }
