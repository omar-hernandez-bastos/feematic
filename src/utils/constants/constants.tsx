export enum ActionsPad {
  CLEAR = 'AC',
  C = 'C',
  DECIMAL = '.',
  CREDIT_CARD = 'ðŸ’³'
}
export const pads = [
  { id: 1, text: 'ðŸ‡¨ðŸ‡´', className: 'col-span-2', color: 'cyan-200' },
  { id: 2, text: 'ðŸ‡ºðŸ‡¸', className: 'col-span-2', color: 'gray-100' },
  { id: 4, text: '7', className: '' },
  { id: 5, text: '8', className: '' },
  { id: 6, text: '9', className: '' },
  { id: 3, text: 'ðŸ‡»ðŸ‡ª', className: '' },
  { id: 7, text: '4', className: '' },
  { id: 8, text: '5', className: '' },
  { id: 9, text: '6', className: '' },
  { id: 3, text: 'ðŸ’³', className: '' },
  { id: 10, text: '1', className: '' },
  { id: 11, text: '2', className: '' },
  { id: 12, text: '3', className: '' },
  { id: 13, text: ActionsPad.C, className: '' },
  { id: 14, text: '0', className: 'col-span-2' },
  { id: 15, text: ActionsPad.DECIMAL, className: '' },
  { id: 16, text: ActionsPad.CLEAR, className: '' }
]

export type CURRENCIES = 'ðŸ‡¨ðŸ‡´' | 'ðŸ‡ºðŸ‡¸' | 'ðŸ‡»ðŸ‡ª'
const currencies = {
  COP: 'ðŸ‡¨ðŸ‡´',
  USD: 'ðŸ‡ºðŸ‡¸',
  VES: 'ðŸ‡»ðŸ‡ª'
}
export const CURRENCIES_ARRAY = [currencies.USD, currencies.COP, currencies.VES]
export { currencies }

export const COP_DEFAULT = 3800
export const VES_DEFAULT = 4.5
