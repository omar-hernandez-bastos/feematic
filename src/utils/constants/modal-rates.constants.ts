const CUSTOM_VES = 'customVes'
const CUSTOM_COP = 'customCop'
const CUSTOM_RATES = 'customRates'
export { CUSTOM_VES, CUSTOM_COP, CUSTOM_RATES }
