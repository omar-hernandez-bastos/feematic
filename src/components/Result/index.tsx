import { Amount } from 'pages/Calculator'
import { useContext } from 'react'
import { formatMoney } from 'utils'
import { COP_DEFAULT, currencies, VES_DEFAULT } from 'utils/constants/constants'
import { RatesContext } from 'utils/contexts/RatesContext'

interface Props {
  amount: Amount
}

const Result = ({ amount }: Props) => {
  const { cop, ves } = useContext(RatesContext)
  const getExchangedAmount = (value: number, currency: string) => {
    if (!cop || !ves)
      return [
        `${formatMoney(value * COP_DEFAULT)} ${currencies.COP}`,
        `${formatMoney(value / (COP_DEFAULT / VES_DEFAULT))} ${currencies.VES}`
      ]
    switch (currency) {
      case currencies.USD:
        return [
          `${formatMoney(value * cop)} ${currencies.COP}`,
          `${formatMoney(value * ves)} ${currencies.VES}`
        ]
      case currencies.COP:
        return [
          `${formatMoney(value / cop)} ${currencies.USD}`,
          `${formatMoney(value / (cop / ves))} ${currencies.VES}`
        ]
      case currencies.VES:
        return [
          `${formatMoney(value * (cop / ves))} ${currencies.COP}`,
          `${formatMoney(value / ves)} ${currencies.USD}`
        ]
      default:
        return [
          `${formatMoney(value * cop)} ${currencies.COP}`,
          `${formatMoney(value * ves)} ${currencies.VES}`
        ]
    }
  }
  return (
    <div className="flex flex-col my-2 text-3xl font-light text-right text-gray-300">
      {getExchangedAmount(+amount.value, amount.currency).map(
        (exchangedAmount, index) => (
          <span key={index}>{exchangedAmount}</span>
        )
      )}
    </div>
  )
}

export default Result
