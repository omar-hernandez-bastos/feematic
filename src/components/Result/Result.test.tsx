import { render } from '@testing-library/react'
import { currencies } from 'utils/constants/constants'
import Result from '.'

describe('Result Test', () => {
  it('Result Component should render correctly', () => {
    const ResultProps = {
      amount: {
        value: '1',
        currency: currencies.USD
      },
      cop: 3800,
      ves: 4.5
    }
    const { container } = render(<Result {...ResultProps} />)
    expect(container).toMatchSnapshot()
  })
  it('Result Component with currency VES', () => {
    const ResultProps = {
      amount: {
        value: '1',
        currency: currencies.VES
      },
      cop: 3800,
      ves: 4.5
    }
    const { container } = render(<Result {...ResultProps} />)
    expect(container).toMatchSnapshot()
  })
  it('Result Component with currency COP', () => {
    const ResultProps = {
      amount: {
        value: '1',
        currency: currencies.COP
      },
      cop: 3800,
      ves: 4.5
    }
    const { container } = render(<Result {...ResultProps} />)
    expect(container).toMatchSnapshot()
  })
  it('Result Component unexpected currency', () => {
    const ResultProps = {
      amount: {
        value: '1',
        currency: 'error'
      },
      cop: 3800,
      ves: 4.5
    }
    const { container } = render(<Result {...ResultProps} />)
    expect(container).toMatchSnapshot()
  })
})
