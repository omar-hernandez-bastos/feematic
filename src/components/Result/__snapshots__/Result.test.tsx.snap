// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Test Result Component should render correctly 1`] = `
<div>
  <div
    class="flex flex-col my-2 text-3xl font-light text-right text-gray-300"
  >
    <span>
      3.800,00 ğŸ‡¨ğŸ‡´
    </span>
    <span>
      4,50 ğŸ‡»ğŸ‡ª
    </span>
  </div>
</div>
`;

exports[`Result Test Result Component unexpected currency 1`] = `
<div>
  <div
    class="flex flex-col my-2 text-3xl font-light text-right text-gray-300"
  >
    <span>
      3.800,00 ğŸ‡¨ğŸ‡´
    </span>
    <span>
      4,50 ğŸ‡»ğŸ‡ª
    </span>
  </div>
</div>
`;

exports[`Result Test Result Component with currency COP 1`] = `
<div>
  <div
    class="flex flex-col my-2 text-3xl font-light text-right text-gray-300"
  >
    <span>
      0,00 ğŸ‡ºğŸ‡¸
    </span>
    <span>
      0,00 ğŸ‡»ğŸ‡ª
    </span>
  </div>
</div>
`;

exports[`Result Test Result Component with currency VES 1`] = `
<div>
  <div
    class="flex flex-col my-2 text-3xl font-light text-right text-gray-300"
  >
    <span>
      844,44 ğŸ‡¨ğŸ‡´
    </span>
    <span>
      0,22 ğŸ‡ºğŸ‡¸
    </span>
  </div>
</div>
`;
